doc/tpid/SonicTPIDSettingHLD1.md

```diff
@@ -0,0 +1,386 @@
+802.1Q-tagged frame. This field is located at the same position as the EtherType/Length field in the untagged frames. This is how a
+switch/router can identify whether the frame received is a tagged or untagged frame. Typical value used for TPID is 0x8100. Some other values
+that have been used are 0x9100, 0x9200 and 0x88A8 (used for Q-in-Q outer TPID). When TPID is present, there is a VLAN ID that follows it.
+The TPID and the VLAN ID combined is typically known as VLAN TAG. The VLAN TAG allows the administrator to separate L2 traffic so that traffic that
+belong to the same VLAN Domain (VLAN ID) will only be forwarded to destination(s) that belong to the same VLAN Domain. This is especially
+useful for L2 flooded/multicast traffic so that interfaces not part of the VLAN Domain will not receive those flooded/multicast packets.
+
+As described earlier, on ingress the TPID configured on the interface serves as a way to detect whether the packet received is Tagged or
+Untagged. On Egress, the TPID configured will be used to "tag" the outgoing packet if the packet needs to be tagged with a VLAN Tag.
+
+![Untagged Tagged Pkt Format](untagged_tagged_pkt_format.png)
+
+Some Network forwarding feature may use some special TPID value such as 0x88A8 for Q-in-Q or PBB (Provider Back Bone). Just want to clarify that
+being able to set the TPID value of an interface to 0x88A8 does not mean that it is able to handle the full feature of Q-in-Q or PBB. It does
+allow the Switch/Router to recognize that there is a VLAN TAG when the outer TPID of the packet received matches that of the TPID configured on
+the ingress interface. When the TPID does not match, the packet is recognized by the receiving interface as untagged. The unrecognized VLAN
+Tag and the rest of that packet are all treated as customer payload and forward it accordingly.
+
+![TPID mismatched Pkt Format](UntaggeTPID_Missmatch.png)
+
+Most Switch/Router HW platforms support configuring TPID on its physical interfaces. A SAI proposal to set the TPID on physical port and LAG has
+already been accepted by the SAI community. This document provides the high-level design for support TPID setting in SONiC.
+
+**SONiC TPID Use case**
+=======================
+
+Currently SONiC Testbed setup uses fanout switches to connect to the testbed server and running the PTF test cases. The fanout switch ports
+connecting to the DUT are configured as access ports. But the PTF test cases also need to test tagged packet functionality. In order to make
+this work, the fanout switch is configured with 802.1Q tunnel on the DUT facing ports side so that tagged packets are not to be used for service
+delimiting purpose. Having the capability for SONiC to configure the port's TPID value will allow SONiC to handle tagged packets in the same
+way. With this TPID configuration capability, these fanout switches can be converted to run in SONiC.
+
+![FanoutSwitch Pkt Flow](FanoutPktFlow.png)
+
+**Linux Kernel TPID Support**

```

Comment:
Hi Gen-Hwa,

In case user configures a non default TPID value (say 0x9100) in such a case what happens to the control packets in the kernel will they be accepted or dropped (ex STP, ARP, ICMP etc). If they are dropped it will break the functionality of these features right ?

Thanks

Reply:
@sandeep-kulambi ,
This HLD was to address the specific use case mentioned in the fanout switch scenario.  It also serves as a starting point for support of 802.q tunneling.  STP BPDU is untagged so it should not matter which TPID is configured.  But if you need to run PVST which needs to be tagged then you will need to re-evaluate why there is a need to change TPID for this port as the kernel does not support non-default TPID values.
Hope this clarify the expectation of this HLD.

Reply:
Thanks Gen-Hwa !!